h/renderers.py

```diff
@@ -2,11 +2,31 @@
 
 from __future__ import unicode_literals
 
+try:
+    from StringIO import StringIO # python 2
+except ImportError:
+    from io import StringIO # python 3
+
 import pyramid.renderers
 
 
 json_sorted_factory = pyramid.renderers.JSON(sort_keys=True)
 
 
+class SVGRenderer(object):
+    def __init__(self, info):
+        pass
+
+    def __call__(self, value, system):
+        request = system.get('request')
+        if request is not None:
```

Comment:
What if `request` _is_ `None`?

Reply:
Dunno! And when is it ever none? This is just copied from the example `CSVRenderer`, so I'd have to investigate a little more

Reply:
Short answer: we don't need this `if`.

All of Pyramid's builtin renderers plus the example `CSVRenderer` from the docs support this case when `request` is `None`. There's no documentation about it that I could find. It's not immediately obvious from the part of the Pyramid source code that calls renderers whether it ever calls them without a request, but I think the answer is no. Pyramid's tests aren't enlightening at all either.

Eventually I asked in the Pyramid IRC channel and the answer I got was that a renderer can be called without a request if some user code does `from pyramid.renderers import render` and then calls Pyramid's `render(...)` directly. If Pyramid doesn't currently have a [threadlocal](https://docs.pylonsproject.org/projects/pyramid//en/latest/narr/threadlocals.html) request (i.e. if we're not currently in a request handling cycle) and our code calls Pyramid's `render(...)` without passing a `request` argument, then Pyramid will end up calling the renderer without a request.

(Or alternatively our own code could just import our `SVGRenderer` class directly and call it without passing it a request). 

You might do this for example in the configuration / startup phase of your app (or any phase outside of request processing - command line interface, worker process...) if you for some reason wanted to render something outside of request processing. The renderers support being used outside of request time like this.

In most outside-of-request-processing situations in h, like CLI and workers etc, I think we actually create a fake request object anyway because so much stuff breaks when there's no request.

But as long as our renderer is only used in rendering responses to requests (as in `@view_config(renderer="svg")` then we don't need to support this.

So we can remove this `if`.